runtime/debugger.c

```diff
@@ -246,6 +246,16 @@ void caml_debugger_init(void)
     sock_addr.s_inet.sin_port = htons(atoi(port));
     sock_addr_len = sizeof(sock_addr.s_inet);
   }
+  /* We completely disable channel locking because:
+     1. The debugger is not compatible with threads, so we don't need locks.
+     2. This file doesn't use locks around channel operations, but
+        function [check_pending] in io.c assumes that the channel is locked
+        and tries to unlock it.
+  */
+  caml_channel_mutex_free = NULL;
+  caml_channel_mutex_lock = NULL;
+  caml_channel_mutex_unlock = NULL;
+  caml_channel_mutex_unlock_exn = NULL;
```

Comment:
I agree this should not make any difference for a single-domain single-threaded program.  Yet, I'd prefer if you would just take the locks on `dbg_in` and `dbg_out` just after they are created, and never release these locks.

(Note to self: investigate why `caml_ml_close_channel` does not lock the channel .)

Reply:
Ah yes, that's a much better solution, thanks.